src/Utility/FieldFunctions.C

```diff
@@ -415,6 +463,46 @@ RealArray FieldsFuncBase::getValues (const Vec3& X)
 }
 
 
+std::vector<Real> FieldsFuncBase::getGradient (const Vec3& X) const
+{
+  const Vec4* x4 = dynamic_cast<const Vec4*>(&X);
+  if (!x4)
+    return {};
+  else
+  {
+    int level = this->findClosestLevel(x4->t);
+    if (level < 0) return {};
+    if (level != currentLevel)
+      if (const_cast<FieldsFuncBase*>(this)->load(fName,bName,level))
+        currentLevel = level;
+
+    Matrix res;
+    field[pidx]->gradFE(ItgPoint(x4->u), res);
+    return res;
```

Comment:
std::move() ? Or will this make no sense here since the return type is different and some implicit casting will take place.

Reply:
precisely